### PR TITLE
SLING-8759 use computeIfAbsent to reduce code size

### PR DESCRIPTION
**Pull request by joerghoh**

### CODE REVIEW

**Review by rombert — CHANGES_REQUESTED** [review]
This does not gracefully degrade when the `ADAPTER_MANAGER` field is not set. Maybe a good opportunity to add a test for that :-) ?

**Comment by joerghoh** [comment]
@rombert thanks for spotting. Added some tests and fixed it.

File: src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+public class SlingAdaptableTest {
+
+
+    SlingAdaptable sut;
```
Comment (rombert) [inline src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java:32]: nit: private members

File: src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+public class SlingAdaptableTest {
+
```
Comment (rombert) [inline src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java:30]: nit: too many blank lines

File: src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        // the 2nd time it has to come out of the cache
+        assertNotNull(sut.adaptTo(AdapterType.class));
+        verify(adapterMgr,times(1)).getAdapter(any(), eq(AdapterType.class));
+
+        assertNull(sut.adaptTo(AdapterType2.class));
+        when (adapterMgr.getAdapter(any(), eq(AdapterType2.class))).thenReturn(new AdapterType2());
+        assertNotNull(sut.adaptTo(AdapterType2.class));
+        assertNotNull(sut.adaptTo(AdapterType.class));
+    }
+
```
Comment (rombert) [inline src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java:65]: nit: too many blank lines

File: src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        sut = new SlingAdaptable() {};
+        adapterMgr = mock(AdapterManager.class);
+    }
+
+    @Test
+    public void testAdaptTo() {
+        assertNull(sut.adaptTo(AdapterType.class));
+        SlingAdaptable.setAdapterManager(adapterMgr);
+        assertNull(sut.adaptTo(AdapterType.class));
+
```
Comment (rombert) [inline src/test/java/org/apache/sling/api/adapter/SlingAdaptableTest.java:46]: nit: extra blank line

**Review by rombert — APPROVED** [review]
LGTM, thanks! I have a couple of style nits, but nothing to re-review for.

**Pull request merged**